src/dataplane-protocol/src/record.rs

```diff
@@ -329,51 +329,51 @@ where
     }
 }
 
-impl<B> Debug for Record<B>
-where
-    B: AsyncBuffer + Debug + Default,
-{
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        writeln!(f, "{:?}", &self.preamble)?;
-        writeln!(f, "{:?}", &self.key)?;
-        writeln!(f, "{:?}", &self.value)?;
-        write!(f, "{:?}", &self.headers)
+impl DefaultRecord {
+    pub fn new<V>(value: V) -> Self
+    where
+        V: Into<Vec<u8>>,
+    {
+        DefaultRecord {
+            value: DefaultAsyncBuffer::new(value.into()),
+            ..Default::default()
+        }
     }
-}
 
-impl<B> From<String> for Record<B>
-where
-    B: From<String> + Default,
-{
-    fn from(value: String) -> Self {
-        Record {
-            value: value.into(),
+    pub fn new_key_value<K, V>(key: K, value: V) -> Self
+    where
+        K: Into<Vec<u8>>,
+        V: Into<Vec<u8>>,
+    {
+        DefaultRecord {
+            key: Some(DefaultAsyncBuffer::new(key.into())),
+            value: DefaultAsyncBuffer::new(value.into()),
             ..Default::default()
         }
     }
 }
 
-impl<B> From<Vec<u8>> for Record<B>
+impl<B> From<String> for Record<B>
```

Comment:
? it was from Vec<u8>

Reply:
Oh there were a handful of them I think. Do you want them all back?

Reply:
Yes, don't need to break old client unnecessary